File: TCDNew-master/src/ui.android/lib/actions/operation/PurchaseTaskAcceptByPapers.dart
import 'package:tiq/actions/data/purchase/PurchaseTaskCreateFromPacs.dart';
import 'package:tiq/actions/ui/purchase/PurchaseTaskContentScan.dart';
import 'package:tiq/actions/ui/purchase/PacHeadScan.dart';
import 'package:tiq/actions/ui/utility/ShowModalMessage.dart';
import 'package:tiq/models/dtos/PacKeyDto.dart';
import 'package:tiq/common/Messages.dart';
import 'package:flutter/material.dart';
import 'package:tiq/services/GStateProvider.dart';

class PurchaseTaskAcceptByPapers {
  Future<void> run(BuildContext context) async {
    var pacHeadList = await PacHeadScan().run(context);
    if (pacHeadList != null) {
      if (pacHeadList.length > 0) {
        final List<PacKeyDto> pacKeys = pacHeadList.cast<PacKeyDto>();
        var purchaseTaskId = await PurchaseTaskCreateFromPacs().run(
            context,
            GStateProvider.instance.settingsFacility.facilityId,
            pacKeys);
        if (purchaseTaskId != null) {
          await PurchaseTaskContentScan().runWithPurchaseTaskId(context, purchaseTaskId);
        }
        else {
          await ShowModalMessage().run(context, Messages.TitleError, 'Не удалось создать новое задание!');
        }
      }
      else {
        await ShowModalMessage().run(context, Messages.TitleError, 'Закупки не выбраны!');
      }
    }
  }
}

File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Actions/Operation/PurchaseTaskAcceptByPapers.cs
using System.Linq;
using Cen.Wms.Client.Actions.Data.Purchase;
using Cen.Wms.Client.Actions.UI.Purchase;
using Cen.Wms.Client.Actions.UI.Utility;
using Cen.Wms.Client.Common;
using Cen.Wms.Client.Models.Dtos;
using Cen.Wms.Client.Services;

namespace Cen.Wms.Client.Actions.Operation
{
    class PurchaseTaskAcceptByPapers
    {
        public static void Run()
        {
            var purchasesHeads = PacHeadScan.Run();

            if (purchasesHeads != null)
            {
                if (purchasesHeads.Count > 0)
                {
                    var purchaseTaskId = PurchaseTaskCreateFromPacs.Run(
                        GStateProvider.Instance.SettingsFacility.FacilityId,
                        purchasesHeads.Select(ph => (PacKeyDto)ph)
                    );
                    if (purchaseTaskId != null)
                    {
                        PurchaseTaskContentScan.Run(purchaseTaskId);
                    }
                    else
                        ShowModalMessage.Run(Messages.TitleError, "Не удалось закрепить закупку за исполнителем!");
                }
                else
                    ShowModalMessage.Run(Messages.TitleError, "Закупки не выбраны!");
            }
        }
    }
}
